transforms/code/license_check/python/src/license_check_transform.py

```diff
@@ -0,0 +1,132 @@
+
+        self.license_check = config.get(LICENSE_CHECK_PARAMS)
+        if not self.license_check:
+            raise ValueError(f"Invalid Argument: cannot create LicenseCheckTransform object.")
+        logger.debug(f"LICENSE_CHECK_PARAMS: {self.license_check}")
+        self.license_column = self.license_check["license_column_name"]
+        allow_no_license = self.license_check["allow_no_license"]
+        licenses = self.license_check["licenses"]
+        deny = self.license_check["deny"]
+        if not deny:
+            self.transformer = AllowLicenseStatusTransformer(
+                license_column=self.license_column,
+                allow_no_license=allow_no_license,
+                licenses=licenses,
+            )
+        else:
+            self.transformer = DenyLicenseStatusTransformer(
+                license_column=self.license_column,
+                allow_no_license=allow_no_license,
+                licenses=licenses,
+            )
+
+    def transform(self, table: pa.Table, file_name: str = None) -> tuple[list[pa.Table], dict]:
+        """
+        Transforms input tables by adding a boolean `license_status` column
+        indicating whether the license is approved/denied.
+        """
+        TransformUtils.validate_columns(table=table, required=[self.license_column])
+        new_table = self.transformer.transform(table)
+        return [new_table], {"n_files": 1}
+
+
+class LicenseCheckTransformConfiguration(TransformConfiguration):
+    def __init__(self):
+        super().__init__(name="license_check", transform_class=LicenseCheckTransform)
+
+    def add_input_params(self, parser: ArgumentParser) -> None:
+        parser.add_argument(
+            "--lc_license_column_name",
+            required=False,
+            type=str,
+            dest="license_column_name",
+            default="contents",
+            help="Name of the column holds the data to process",
+        )
+        parser.add_argument(
+            "--lc_allow_no_license",
+            required=False,
+            dest="allow_no_license",
+            action="store_true",
+            default=False,
+            help="allow entries with no associated license (default: false)",
+        )
+        parser.add_argument(
+            "--lc_licenses_file",
+            required=True,
+            dest="licenses_file",
+            type=str,
+            help="S3 or local path to allowed/denied licenses JSON file",
+        )
+        parser.add_argument(
+            "--lc_deny_licenses",
+            required=False,
+            dest="deny",
+            action="store_true",
+            default=False,
+            help="allow all licences except those in licenses_file (default: false)",
+        )
+
+    def apply_input_params(self, args: Namespace) -> bool:
+        dargs = vars(args)
+        # Read licenses from allow-list or deny-list
+        data_access = DataAccessFactory("lc_", False).create_data_access()
```

Comment:
This is in the wrong place I think. You need to create the DAF in add_input_params(), and then daf.add_input_params(parser).  THis way the cli will support the params used to load the file (s3 or local).  See

https://github.com/IBM/data-prep-kit/blob/76293fab3e92a7e714812400ed750eac2066bacf/transforms/code/proglang_select/python/src/proglang_select_transform.py#L148

Reply:
done